lib/dynamoid/adapter.rb

```diff
@@ -19,7 +19,7 @@ def tables
       if !@tables_.value
         @tables_.swap{|value, args| benchmark('Cache Tables') {list_tables}}
       end
-      @tables_.value
+      @tables_.value || []
```

Comment:
Very interesting report. In my case, .value was nil and it was causing later lines of code doing a tables.include? to fail if I had no tables initially. Is there a reason why it would be bad to default to an empty array here? There's almost no performance cost and obviously there are edge cases where .value is nil. I leave it up to the community to decide. I'll keep using my fork if you want to get rid of that change since this works great for me.